### ISSUE TITLE
Aurora missing their first shot

### ISSUE THREAD

**Issue by JaggedAppliance** [FAForever/fa#2924]
This is a pretty bad issue, aurora are missing their first shot extremely often, after the first shot they have no trouble with accuracy. Any insight into why this may be happening would be appreciated.

**Comment by wodzu93** [comment]
"FiringTolerance" parameter. It tells how much barrel can be off-target before shot is allowed, in degrees. Auroras have Firing Tolerance of 2, so 2 degrees off-target first shot is common if unit wasn't facing the target already.

**Comment by Petricpwnz** [comment]
So just in case you are wondering why your PR isn't being merged, or someone else wants to dig deeper, I've done about an hour of testing like a week or 2 ago on this issue and don't think firing tolerance is the culprit here or at least not the only one. Firstly the issue is extremely inconsistent, as from the balance call jagged said it happened to him in like every aeon game he played/saw but neither me nor turin were able to reproduce this in testing. What I found however was that it is certainly way more likely to miss the first shot when it has no radar as it doesn't start tracking a unit before it enters it's range. Another issue is that when enemy unit passes by at close proximity aurora has to turn before it can reload multiple times in a row, that makes it not shoot for a while, effectively losing a good chunk of its dps. I feel like the misses are largely the same kind of misses you see on t1 pd shooting units moving diagonally tbh.
So to summarize there are 2 issues:

- Missing (1st?) shot, most often without radar but maybe with as well? Inconsistent in trying to reproduce
- Losing DPS while having to turn a lot.

For the 1st issue what i found helped a lot was removing muzzle charge delay but i dont really like how it feels and also it removed the shooting sound and i was too lazy to even check if its a 1 line fix or something bigger. Anyway, I think the issue is more complex and there are probably multiple reasons and situations where it misses.
For the 2nd have to play with the turret or body turn rate, although quick value tweaks somehow completely broke the aurora aim so I kinda dropped it at that.

**Comment by wodzu93** [comment]
`UseFiringSolutionInsteadOfAimBone = true,`
I had some aiming issues with a modded unit where it was consistently missing moving units (unit was shooting too much ahead of the target). Adding above line to the weapon fixed the problem. Maybe it is worth trying it here.

**Comment by Petricpwnz** [comment]
From my limited testing this seems to fix the 1st problem or at least help with it. I would revert the firing tolerance tho, it seemed to work fine without a change to it and I imagine it could maybe even make the 2nd problem worse.

**Comment by wodzu93** [comment]
Sure, I'll do that.
